infrastructure/environments/dplplat01/configuration/ingress-nginx/ingres-nginx-values.template.yaml

```diff
@@ -13,6 +13,10 @@ controller:
     # See https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/annotations/#custom-max-body-size
     # and https://github.com/uselagoon/lagoon-images/blob/main/images/nginx/nginx.conf#L81
     proxy-body-size: "2048m"
+    # Default client_header_buffer_size is 1k in nginx, which results in 502
+    # errors for some applications, eg. large cookie headers.
+    # See https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/configmap/#client-header-buffer-size
+    client-header-buffer-size: "8k"
```

Comment:
While I do not doubt this might do the job. Are you sure that it is the only setting needed?
According to this post, a guy managed to fix the problem with some other settings:
https://community.auth0.com/t/next-js-app-returns-502-bad-gateway-for-api-auth-me-after-a-successful-login/135909/3

Reply:
Same story here:
https://stackoverflow.com/questions/57294212/502-bad-gateway-error-upstream-sent-too-big-header-while-reading-response-head

Reply:
And here:
https://unix.stackexchange.com/a/605614

Reply:
I'm not completely sure, but I think the `proxy_*` headers are only relevant with `proxy_pass` traffic, ie. when speaking with PHP-FPM.

In this case, we are trying to affect the HTTP header of an incoming client request, which is why I think [`client_header_buffer_size`](https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/configmap/#client-header-buffer-size) is most likely.

Reply:
Two of the links I posted refers to a setup with nginx in front of a running node application. Eg. [this](https://unix.stackexchange.com/questions/605467/how-to-handle-a-too-large-cookie-causing-nginx-to-return-a-502) one.
Wouldn't that always be in a proxy relation like the one we have with PHP-FPM?

Reply:
Possibly. This is the generated block of config from `nginx.conf` in a ingress-nginx container:

```lua

	## start server node.pr-1757.dpl-cms.dplplat01.dpl.reload.dk
	server {
		server_name node.pr-1757.dpl-cms.dplplat01.dpl.reload.dk ;
		
		listen 80  ;
		listen [::]:80  ;
		listen 443  ssl http2 ;
		listen [::]:443  ssl http2 ;
		
		set $proxy_upstream_name "-";
		
		ssl_certificate_by_lua_block {
			certificate.call()
		}
		
		# Custom code snippet configured for host node.pr-1757.dpl-cms.dplplat01.dpl.reload.dk
		add_header X-Robots-Tag "noindex, nofollow";
		
		location / {
			
			set $namespace      "dpl-cms-pr-1757";
			set $ingress_name   "node";
			set $service_name   "node";
			set $service_port   "http";
			set $location_path  "/";
			set $global_rate_limit_exceeding n;
			
			rewrite_by_lua_block {
				lua_ingress.rewrite({
					force_ssl_redirect = false,
					ssl_redirect = false,
					force_no_ssl_redirect = false,
					preserve_trailing_slash = false,
					use_port_in_redirects = false,
					global_throttle = { namespace = "", limit = 0, window_size = 0, key = { }, ignored_cidrs = { } },
				})
				balancer.rewrite()
				plugins.run()
			}
			
			# be careful with `access_by_lua_block` and `satisfy any` directives as satisfy any
			# will always succeed when there's `access_by_lua_block` that does not have any lua code doing `ngx.exit(ngx.DECLINED)`
			# other authentication method such as basic auth or external auth useless - all requests will be allowed.
			#access_by_lua_block {
			#}
			
			header_filter_by_lua_block {
				lua_ingress.header()
				plugins.run()
			}
			
			body_filter_by_lua_block {
				plugins.run()
			}
			
			log_by_lua_block {
				balancer.log()
				
				monitor.call()
				
				plugins.run()
			}
			
			port_in_redirect off;
			
			set $balancer_ewma_score -1;
			set $proxy_upstream_name "dpl-cms-pr-1757-node-http";
			set $proxy_host          $proxy_upstream_name;
			set $pass_access_scheme  $scheme;
			
			set $pass_server_port    $server_port;
			
			set $best_http_host      $http_host;
			set $pass_port           $pass_server_port;
			
			set $proxy_alternative_upstream_name "";
			
			client_max_body_size                    2048m;
			
			proxy_set_header Host                   $best_http_host;
			
			# Pass the extracted client certificate to the backend
			
			# Allow websocket connections
			proxy_set_header                        Upgrade           $http_upgrade;
			
			proxy_set_header                        Connection        $connection_upgrade;
			
			proxy_set_header X-Request-ID           $req_id;
			proxy_set_header X-Real-IP              $remote_addr;
			
			proxy_set_header X-Forwarded-For        $remote_addr;
			
			proxy_set_header X-Forwarded-Host       $best_http_host;
			proxy_set_header X-Forwarded-Port       $pass_port;
			proxy_set_header X-Forwarded-Proto      $pass_access_scheme;
			proxy_set_header X-Forwarded-Scheme     $pass_access_scheme;
			
			proxy_set_header X-Scheme               $pass_access_scheme;
			
			# Pass the original X-Forwarded-For
			proxy_set_header X-Original-Forwarded-For $http_x_forwarded_for;
			
			# mitigate HTTPoxy Vulnerability
			# https://www.nginx.com/blog/mitigating-the-httpoxy-vulnerability-with-nginx/
			proxy_set_header Proxy                  "";
			
			# Custom headers to proxied server
			
			proxy_connect_timeout                   5s;
			proxy_send_timeout                      60s;
			proxy_read_timeout                      60s;
			
			proxy_buffering                         off;
			proxy_buffer_size                       4k;
			proxy_buffers                           4 4k;
			
			proxy_max_temp_file_size                1024m;
			
			proxy_request_buffering                 on;
			proxy_http_version                      1.1;
			
			proxy_cookie_domain                     off;
			proxy_cookie_path                       off;
			
			# In case of errors try the next upstream server before returning an error
			proxy_next_upstream                     error timeout;
			proxy_next_upstream_timeout             0;
			proxy_next_upstream_tries               3;
			
			proxy_pass http://upstream_balancer;
			
			proxy_redirect                          off;
			
		}
		
	}
```

So some of the proxy directives are already being configured even if we do not set them explicitly in the ConfigMap.

Reply:
According to [this](https://unix.stackexchange.com/a/605614) post, we should increase: proxy_buffer_size and proxy_buffers. Do we have access to manipulate those values?